Sources/Commands/Workspace.swift

```diff
@@ -301,6 +304,10 @@ public class Workspace {
         try localFileSystem.createDirectory(repositoriesPath, recursive: true)
         try localFileSystem.createDirectory(checkoutsPath, recursive: true)
 
+        // Create a file lock and immediately try to aquire the lock before restoring the previous state.
+        fileLock = FileLock(name: "swiftpm-workspace-lock", cachePath: self.dataPath)
+        try fileLock.lock()
```

Comment:
What would be a good number? This could be pretty long for eg when you're fetching dependencies.